src/WebJobs.Script.WebHost/Security/SecretManager.cs

```diff
@@ -446,5 +425,76 @@ private void OnSecretsChanged(object sender, SecretsChangedEventArgs e)
 
             await _repository.PurgeOldSecretsAsync(currentFunctions, traceWriter, logger);
         }
+
+        private async Task<HostSecrets> LoadHostSecretsWithRetryAsync()
+        {
+            int currentRetry = 0;
+            for (; ;)
```

Comment:
With the exception filter mentioned below, you can define and increment your counter in your `for` statement